newIDE/app/src/MainFrame/index.js

```diff
@@ -1522,548 +1573,545 @@ class MainFrame extends React.Component<Props, State> {
+        toggleProjectManager={toggleProjectManager}
+        exportProject={() => openExportDialog(true)}
+        requestUpdate={props.requestUpdate}
+        simulateUpdateDownloaded={simulateUpdateDownloaded}
+        simulateUpdateAvailable={simulateUpdateAvailable}
+      />
+      <ClosableTabs hideLabels={!!props.integratedEditor}>
+        {getEditors(state.editorTabs).map((editorTab, id) => {
+          const isCurrentTab =
+            getCurrentTabIndex(state.editorTabs) === id;
+          return (
+            <ClosableTab
+              label={editorTab.label}
+              key={editorTab.key}
+              active={isCurrentTab}
+              onClick={() => _onChangeEditorTab(id)}
+              onClose={() => _onCloseEditorTab(editorTab)}
+              onCloseOthers={() => _onCloseOtherEditorTabs(editorTab)}
+              onCloseAll={_onCloseAllEditorTabs}
+              onActivated={() => _onEditorTabActive(editorTab)}
+              closable={editorTab.closable}
+            />
+          );
+        })}
+      </ClosableTabs>
+      {getEditors(state.editorTabs).map((editorTab, id) => {
+        const isCurrentTab = getCurrentTabIndex(state.editorTabs) === id;
+        return (
+          <TabContentContainer key={editorTab.key} active={isCurrentTab}>
+            <ErrorBoundary>{editorTab.render(isCurrentTab)}</ErrorBoundary>
+          </TabContentContainer>
+        );
+      })}
+      <LoaderModal show={showLoader} />
+      <HelpFinder
+        open={helpFinderDialogOpen}
+        onClose={() => openHelpFinderDialog(false)}
+      />
+      <Snackbar
+        open={state.snackMessageOpen}
+        autoHideDuration={3000}
+        onClose={_closeSnackMessage}
+        ContentProps={{
+          'aria-describedby': 'snackbar-message',
+        }}
+        message={<span id="snackbar-message">{state.snackMessage}</span>}
+      />
+      {!!renderExportDialog &&
+        state.exportDialogOpen &&
+        renderExportDialog({
+          onClose: () => openExportDialog(false),
+          onChangeSubscription: () => {
+            openExportDialog(false);
+            openSubscription(true);
+          },
+          project: state.currentProject,
+        })}
+      {!!renderCreateDialog &&
+        state.createDialogOpen &&
+        renderCreateDialog({
+          open: state.createDialogOpen,
+          onClose: () => openCreateDialog(false),
+          onOpen: (storageProvider, fileMetadata) => {
+            openCreateDialog(false);
+            // eslint-disable-next-line
+            useStorageProvider(storageProvider)
+              .then(() => openFromFileMetadata(fileMetadata))
+              .then(() => openSceneOrProjectManager());
+          },
+          onCreate: (project, storageProvider, fileMetadata) => {
+            openCreateDialog(false);
+            // eslint-disable-next-line
+            useStorageProvider(storageProvider)
+              .then(() => loadFromProject(project, fileMetadata));
```

Comment:
Is there a reason not for adding `.then(() => openSceneOrProjectManager());` like before? (is it handled in a hook?)

Reply:
Okay, about this, I had some doubts too. This is the only instance where `loadFromProject()` is called and it used to resolve the promise inside the callback of setState.

> is it handled in a hook?

Yes, I had to extract the callback and put the code inside `useEffect`. But we know that this would basically mean we can't chain it anymore. Since there was only one instance of this function getting called, I handled it inside the hook.

Now, I'm unsure whether `loadFromProject` is called anywhere else other than MainFrame, because this would be an issue.

Reply:
I think that's the only place where it's called: https://github.com/4ian/GDevelop/search?q=loadFromProject&unscoped_q=loadFromProject